pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotSegmentRestletResource.java

```diff
@@ -19,255 +19,409 @@
+    for (String tableNameWithType : tableNamesWithType) {
+      Map<String, Object> resultForTable = new LinkedHashMap<>();
+      resultForTable.put("tableName", tableNameWithType);
+      resultForTable.put("serverToSegmentsMap", _pinotHelixResourceManager.getServerToSegmentsMap(tableNameWithType));
+      resultList.add(resultForTable);
     }
-    return toggleStateInternal(tableName, state, tableType, null, _pinotHelixResourceManager);
+    return resultList;
   }
 
+  @Deprecated
   @GET
-  @Path("tables/{tableName}/segments/{segmentName}")
+  @Path("tables/{tableName}/segments")
   @Produces(MediaType.APPLICATION_JSON)
-  @ApiOperation(value = "Lists metadata or toggles state of a segment", notes = "Toggles segment state if 'state' is specified in query param, otherwise lists segment metadata")
-  public String toggleStateOrListMetadataForOneSegment(
+  @ApiOperation(value = "Get a map from server to segments hosted by the server (deprecated, use 'GET /segments/{tableName}/servers' instead)", notes = "Get a map from server to segments hosted by the server (deprecated, use 'GET /segments/{tableName}/servers' instead)")
+  public List<Map<String, String>> getServerToSegmentsMapDeprecated1(
       @ApiParam(value = "Name of the table", required = true) @PathParam("tableName") String tableName,
-      @ApiParam(value = "Name of segment", required = true) @PathParam("segmentName") @Encoded String segmentName,
-      @ApiParam(value = "enable|disable|drop", required = false) @QueryParam("state") String stateStr,
-      @ApiParam(value = "realtime|offline", required = false) @QueryParam("type") String tableTypeStr) {
-    segmentName = URIUtils.decode(segmentName);
-    // segmentName will never be null,otherwise we would reach the method toggleStateOrListMetadataForAllSegments()
-    CommonConstants.Helix.TableType tableType = Constants.validateTableType(tableTypeStr);
-    StateType stateType = Constants.validateState(stateStr);
-    if (stateStr == null) {
-      // This is a list metadata operation
-      return listSegmentMetadataInternal(tableName, segmentName, tableType);
-    } else {
-      // We need to toggle state
-      if (tableType == null) {
-        List<String> realtimeSegments =
-            _pinotHelixResourceManager.getSegmentsFor(TableNameBuilder.REALTIME.tableNameWithType(tableName));
-        List<String> offlineSegments =
-            _pinotHelixResourceManager.getSegmentsFor(TableNameBuilder.OFFLINE.tableNameWithType(tableName));
-        if (realtimeSegments.contains(segmentName)) {
-          tableType = CommonConstants.Helix.TableType.REALTIME;
-        } else if (offlineSegments.contains(segmentName)) {
-          tableType = CommonConstants.Helix.TableType.OFFLINE;
-        }
-      }
-      return toggleStateInternal(tableName, stateType, tableType, segmentName, _pinotHelixResourceManager);
+      @ApiParam(value = "MUST be null") @QueryParam("state") String stateStr,
```

Comment:
required = false?

Reply:
required is false by default (IDE will even warn about explicit required = false)